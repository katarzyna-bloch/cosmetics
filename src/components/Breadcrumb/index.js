import React from 'react'
import styled from 'styled-components'
import { Link as RouterLink } from 'react-router-dom'
import PropTypes from 'prop-types'

const Wrapper = styled.div`
  font-family: Arial, Helvetica, sans-serif;
  font-size: 15px;
`

const Element = styled.span`
  color: gray;
  &:not(:last-child):after {
    padding: 8px;
    content: "/";
  }
`

const Link = styled(RouterLink)`
  color: inherit;
  text-decoration: none;
  &:hover {
    color: lightblue;
  }
`

const Label = styled.span``

const Breadcrumb = ({ values }) => {
  return (
    <Wrapper>
      <Element>
        <Link to="/">Home</Link>
      </Element>
      {values.map((value) => (
        <Element key={value.label}>
          {value.link ? (
            <Link to={value.link}>{value.label}</Link>
          ) : (
            <Label>{value.label}</Label>
          )}
        </Element>
      ))}
    </Wrapper>
  )
}

Breadcrumb.propTypes = {
  values: PropTypes.array.isRequired,
}

export default Breadcrumb
