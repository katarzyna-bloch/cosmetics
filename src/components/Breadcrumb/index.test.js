import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Breadcrumb from '.'

const props = {
  values: [
    { label: "Domowe SPA", link: "/categories/home-spa" },
    { label: "Cytrusowy Peeling Enzymatyczny w Å»elu" }
  ],
}

it('should render correctly', () => {
  const wrapped = shallow(<Breadcrumb {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
