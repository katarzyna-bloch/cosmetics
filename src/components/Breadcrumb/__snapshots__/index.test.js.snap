// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div>
  <styled.span>
    <Styled(Link)
      to="/"
    >
      Home
    </Styled(Link)>
  </styled.span>
  <styled.span
    key="Domowe SPA"
  >
    <Styled(Link)
      to="/categories/home-spa"
    >
      Domowe SPA
    </Styled(Link)>
  </styled.span>
  <styled.span
    key="Cytrusowy Peeling Enzymatyczny w Żelu"
  >
    <styled.span>
      Cytrusowy Peeling Enzymatyczny w Żelu
    </styled.span>
  </styled.span>
</styled.div>
`;
